fairlearn/datasets/_fetch_compas.py

```diff
@@ -0,0 +1,139 @@
+        a pandas DataFrame or Series depending on the number of target_columns.
+        The Bunch will contain a ``frame`` attribute with the target and the
+        data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
+        DataFrames or Series as describe above.
+
+    return_X_y : boolean, default=False.
+        If True, returns ``(data.data, data.target)`` instead of a Bunch
+        object.
+
+    warn : boolean, default=True.
+        If True, it raises an extra warning to make users aware of the unfairness
+        aspect of this dataset.
+
+
+    Returns
+    -------
+    dataset : :class:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the following attributes.
+
+        data : ndarray, shape (5278, 13)
+            Each row corresponding to the 13 feature values in order.
+            If ``as_frame`` is True, ``data`` is a pandas object.
+        target : numpy array of shape (5278,)
+            Each value indicates whether a defendant was charged again within two years after the
+            current charge.
+            If ``as_frame`` is True, ``target`` is a pandas object.
+        feature_names : list of length 13
+            Array of ordered feature names used in the dataset.
+        DESCR : string
+            Description of the COMPAS dataset.
+
+    (data, target) : tuple of (numpy.ndarray, numpy.ndarray) or (pandas.DataFrame, pandas.Series)
+        if ``return_X_y`` is True and ``as_frame`` is False
+
+    (data, target) : tuple of (pandas.DataFrame, pandas.Series)
+        if ``return_X_y`` is True and ``as_frame`` is True
+
+    See Also
+    --------
+    fairlearn.datasets.fetch_boston: Boston housing dataset (regression)
+    fairlearn.datasets.fetch_adult: UCI Adult dataset (binary classification)
+
+    Notes
+    -----
+    This dataset consists of 5278 samples and 13 features.
+
+    - During pre-processing only instances of defendants of the two most common races were kept,
+      which means that `race_African-American` and `race_Caucasian` are perfectly correlated.
+    - During pre-processing charges for ordinary traffic offenses were removed, which means that
+      c_charge_degree_F` and `c_charge_degree_M` are perfectly correlated.
+    - The features `age_cat_25-45`, `age_cat_Greaterthan45` and `age_cat_Lessthan25` are included
+      to allow linear classifiers to be non-linear in age.
+
+
+    References
+    ----------
+    .. [1] https://github.com/propublica/compas-analysis/
+    .. [2] https://www.openml.org/d/42193
+    .. [3] https://www.propublica.org/article/how-we-analyzed-the-compas-recidivism-algorithm/
+
+    """
+    if warn:
+        msg = "You are about to use a dataset with known fairness issues."
+        warnings.warn(DataFairnessWarning(msg))
+    if not data_home:
+        data_home = pathlib.Path().home() / _DOWNLOAD_DIRECTORY_NAME
+    return fetch_openml(
+        data_id=42193,
```

Comment:
Is the other version of this dataset mentioned in the discussion also part of openml, or would that require new download code?

Reply:
Not that I'm aware, but if we manage to get it in the right format we can just upload it :)

Reply:
The referenced one was actually uploaded by me :-D
I remember it being a huge pain since I had to convert it to some obscure format.